### DISCUSSION TITLE
ERROR - XNVCtrl failed: XOpenDisplay failed; is $DISPLAY set?

### DISCUSSION THREAD

**Discussion by kevinzezel** [foucault/nvfancontrol#47] [General]
Hi,

I use my computer as a server and I don't have a monitor connected to it. is it possible to run?

Regards,
Kevin

**Comment by foucault** [comment]
Unfortunately the same caveats with wanting to run a headless nvidia gpu apply here as well. This is not really a bug. You need a running X Server to do anything with XNVCtrl. An obvious solution is to buy a dummy displayport/hdmi plug (that essentially emulates a monitor) and run X on that "display". There is also the [Persistence Daemon](https://docs.nvidia.com/deploy/driver-persistence/index.html#persistence-daemon) but I'm not entirely certain if this allows XNVCtrl to initialise.
